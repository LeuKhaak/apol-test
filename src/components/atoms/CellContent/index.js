import React from "react";
import * as styles from "./styles.module.scss";
import cn from "classnames";

function CellContent({ content, cellStyle, mode, id }) {
  const classCell = cn(styles.cell, styles[cellStyle]);
  return (
    <div
      className={classCell}
      style={{ display: mode === "edite" ? "none" : "" }}
      id={id}
    >
      {content}
    </div>
  );
}

export default CellContent;
