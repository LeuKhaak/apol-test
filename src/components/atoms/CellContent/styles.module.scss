@import "../../../assets/styles/styles.scss";
.cell {
  padding: 0.5rem;
  min-height: 1rem;
  font-weight: 400;
  border: solid 0.1rem $greyBorderColor;
  background-color: $bgcLightGrey;
  overflow: auto;
}
.cellTitle,
.number {
  border: none;
  background: none;
}
.cellTitle {
  font-weight: bold;
}
