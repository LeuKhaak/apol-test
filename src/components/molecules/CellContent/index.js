import React from "react";
import * as styles from "./styles.module.scss";
import cn from "classnames";

function CellContent({ content, cellStyle, cellContentStyle, id }) {
  const classCell = cn(styles.cell, styles[cellStyle]);
  return (
    <p className={classCell} id={id}>
      {content}
      <span
        className={styles.cellSimbol}
        style={{ display: cellContentStyle === "simbol" ? "flex" : "" }}
      >
        &#8595;
      </span>
    </p>
  );
}

export default CellContent;
