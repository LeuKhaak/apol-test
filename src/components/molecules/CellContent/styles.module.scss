@import "../../../assets/styles/styles.scss";
.cell {
  padding: 0.5rem 0.25rem;
  min-height: 1rem;
  font-weight: 400;
  border: solid 0.1rem $greyBorderColor;
  background-color: $bgcLightGrey;
  overflow: auto;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  @include tabletSmallSecond {
    flex-direction: column;
  }
  .cellSimbol {
    display: none;
  }
  @include desktop {
    padding: 0.5rem 0.25rem;
    font-size: 0.75rem;
  }
  @include tabletLandscape {
    padding: 0.5rem 0.15rem;
  }
  @include mobileSmallSecond {
    padding: 0.5rem 0.1rem;
    font-size: 0.65rem;
  }
}
.cellTitle,
.cellTitleActions {
  border: none;
  background: none;
  font-weight: bold;
  @include tabletLandscape {
    font-weight: 400;
  }
  @include tabletSmallSecond {
    min-height: 2.6rem;
    padding: 0 0.15rem;
  }
}
.number {
  border: none;
  background: none;
}
