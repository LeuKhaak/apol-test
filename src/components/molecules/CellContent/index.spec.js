import React from "react";
import { shallow } from "enzyme";
import CellContent from ".";

let component = null;
const setComponent = (props) => shallow(<CellContent {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("CellContent testing:", () => {
  test("is CellContent contain text", () => {
    component = setComponent({
      content: "testText",
    });
    expect(component).toMatchSnapshot();
  });
  test("is CellContent has sortIcon", () => {
    component = setComponent({
      content: "testText",
      cellContentStyle: "simbol",
    });
    expect(component).toMatchSnapshot();
  });
});
