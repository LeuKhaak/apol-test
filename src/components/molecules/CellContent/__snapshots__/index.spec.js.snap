// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellContent testing: is CellContent contain text 1`] = `
<p
  className=""
>
  testText
  <span
    style={
      Object {
        "display": "",
      }
    }
  >
    ↓
  </span>
</p>
`;

exports[`CellContent testing: is CellContent has sortIcon 1`] = `
<p
  className=""
>
  testText
  <span
    style={
      Object {
        "display": "flex",
      }
    }
  >
    ↓
  </span>
</p>
`;
